### PR TITLE
[IMPROVE] Increase the verbosity of Omnichannel routing system debugging outputs

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

Pls let me know if there are any other places where we can add more logs 😬

### CODE REVIEW

File: app/livechat/server/lib/Livechat.js [hunk 1]
```diff
@@ -176,13 +189,15 @@ export const Livechat = {
 
 	async sendMessage({ guest, message, roomInfo, agent }) {
 		const { room, newRoom } = await this.getRoom(guest, message, roomInfo, agent);
+		Livechat.logger.debug(`Attempting to send a message by visitor ${ guest._id } on room ${ room._id }`);
```
Comment (renatobecker) [inline app/livechat/server/lib/Livechat.js:192]: We don't need to log this event.

File: app/livechat/server/lib/Livechat.js [hunk 2]
```diff
@@ -176,13 +189,15 @@ export const Livechat = {
...
 		if (guest.name) {
 			message.alias = guest.name;
 		}
 		return _.extend(sendMessage(guest, message, room), { newRoom, showConnecting: this.showConnecting() });
 	},
 
 	updateMessage({ guest, message }) {
+		Livechat.logger.debug(`Attempting to update a message by visitor ${ guest._id }`);
```
Comment (renatobecker) [inline app/livechat/server/lib/Livechat.js:200]: We don't need to log this event.

File: app/livechat/server/lib/QueueManager.js [hunk 3]
```diff
@@ -4,20 +4,26 @@ import { Match, check } from 'meteor/check';
 import { LivechatRooms, LivechatInquiry, Users } from '../../../models/server';
 import { checkServiceStatus, createLivechatRoom, createLivechatInquiry } from './Helper';
 import { callbacks } from '../../../callbacks/server';
+import { Logger } from '../../../logger';
 import { RoutingManager } from './RoutingManager';
 
+const logger = new Logger('QueueMananger');
 
 export const queueInquiry = async (room, inquiry, defaultAgent) => {
+	logger.debug(`Delegating inquiry with id ${ inquiry._id } to agent ${ defaultAgent?._id }`);
```
Comment (renatobecker) [inline app/livechat/server/lib/QueueManager.js:13]: I think this method should be called after:
`const inquiryAgent = RoutingManager.delegateAgent(defaultAgent, inquiry);`

File: app/livechat/server/lib/RoutingManager.js [hunk 4]
```diff
@@ -40,19 +45,23 @@ export const RoutingManager = {
...
 	async getNextAgent(department, ignoreAgentId) {
+		logger.debug(`Getting next available agent with method ${ this.name }`);
 		return this.getMethod().getNextAgent(department, ignoreAgentId);
 	},
 
 	async delegateInquiry(inquiry, agent, options = {}) {
 		const { department, rid } = inquiry;
+		logger.debug(`Attempting to delegate inquiry ${ inquiry._id }`);
 		if (!agent || (agent.username && !Users.findOneOnlineAgentByUserList(agent.username) && !allowAgentSkipQueue(agent))) {
 			agent = await this.getNextAgent(department);
```
Comment (renatobecker) [inline app/livechat/server/lib/RoutingManager.js:56]: There's a log missing after this line.

**Review by renatobecker — CHANGES_REQUESTED** [review]
I'm missing logs on https://github.com/RocketChat/Rocket.Chat/blob/develop/ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js

**Review by renatobecker — APPROVED** [review]

**Review by renatobecker — APPROVED** [review]

File: ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js [hunk 5]
```diff
@@ -245,5 +260,12 @@ const queueWorker = {
...
 settings.onload('Livechat_Routing_Method', function() {
-	RoutingManager.getConfig().autoAssignAgent ? queueWorker.start() : queueWorker.stop();
+	const routingSupportsAutoAssign = RoutingManager.getConfig().autoAssignAgent;
+	logger.queue.debug(`Routing method ${ RoutingManager.methodName } supports auto assignment: ${ routingSupportsAutoAssign }. ${
+		routingSupportsAutoAssign
+			? 'Starting'
+			: 'Stopping'
+	} queue`);
+
+	routingSupportsAutoAssign ? queueWorker.start() : queueWorker.stop();
```
Comment (pierre-lehnen-rc) [inline ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js:270]: This queueWorker is running even when omnichannel is disabled and it's generating a lot of useless debug logs every second.

**Pull request merged**